src/transformers/models/gemma/modeling_gemma.py

```diff
@@ -101,16 +101,19 @@ def __init__(self, dim, max_position_embeddings=2048, base=10000, device=None):
         self.base = base
         self.register_buffer("inv_freq", None, persistent=False)
 
+    @torch.no_grad()
```

Comment:
Are you sure we don't need the gradient? cos/sin are later used so should be taken into account for e.g. `v_proj` no?

Reply:
They are just added it's a constant, gradients don't propagate to the sin or cos or anything inside that module

Reply:
Oh so cos/sin are used for RoPE, but we dont need gradients for cos and sin itself. But tbh I have not yet tested no_grad itself for training, so will have to do that to confirm